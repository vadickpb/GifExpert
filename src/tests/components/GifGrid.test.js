import React from 'react';
import { useFetchGif } from '../../hooks/useFetchGif';
const { shallow } = require("enzyme")
const { GifGrid } = require("../../components/GifGrid")
jest.mock('../../hooks/useFetchGif');

describe('Pruebas en <GifGrid.js />', () => {
    
    const category = 'One Punch';

    test('Debe mostrarse correctamente', () => {

        useFetchGif.mockReturnValue({
            data: [], 
            loading: true
        })

        const wrapper = shallow(<GifGrid category =  {category} />);
        expect(wrapper).toMatchSnapshot();

    })

    test('deben mostrarse los items cuando se cargan las imagenes en UseFetchGifs ', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://cualquiercosa.com',
            title: 'Cualquier cosa'
        }]

        useFetchGif.mockReturnValue({
            data: gifs, 
            loading: false
        })
        const wrapper = shallow(<GifGrid category =  {category} />);
        
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
        
    })
    


    
})
