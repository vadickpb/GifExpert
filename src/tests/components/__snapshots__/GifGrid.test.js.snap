// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid.js /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Cargando..
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid.js /> deben mostrarse los items cuando se cargan las imagenes en UseFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://cualquiercosa.com"
    />
  </div>
</Fragment>
`;
